packages/datagateway-table/src/breadcrumbs.component.test.tsx

```diff
@@ -0,0 +1,598 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import { createMount } from '@material-ui/core/test-utils';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { MemoryRouter } from 'react-router';
+
+import { initialState } from './state/reducers/dgtable.reducer';
+import { StateType } from './state/app.types';
+import { createLocation } from 'history';
+
+import PageBreadcrumbs from './breadcrumbs.component';
+import axios from 'axios';
+import { ReactWrapper } from 'enzyme';
+
+jest.mock('loglevel');
+
+// The generic routes to test.
+const genericRoutes = {
+  investigations: '/browse/investigation',
+  datasets: '/browse/investigation/1/dataset',
+  datafiles: '/browse/investigation/1/dataset/1/datafile',
+};
+
+// The ISIS routes to test.
+const ISISRoutes = {
+  instruments: '/browse/instrument',
+  facilityCycles: '/browse/instrument/1/facilityCycle',
+  investigations: '/browse/instrument/1/facilityCycle/1/investigation',
+  datasets: '/browse/instrument/1/facilityCycle/1/investigation/1/dataset',
+  datafiles:
+    '/browse/instrument/1/facilityCycle/1/investigation/1/dataset/1/datafile',
+};
+
+// The DLS routes to test.
+const DLSRoutes = {
+  proposals: '/browse/proposal',
+  investigations: '/browse/proposal/INVESTIGATION 1/investigation',
+  datasets: '/browse/proposal/INVESTIGATION 1/investigation/1/dataset',
+  datafiles:
+    '/browse/proposal/INVESTIGATION 1/investigation/1/dataset/1/datafile',
+};
+
+describe('PageBreadcrumbs - Snapshot Tests (Generic, DLS, ISIS)', () => {
+  let mount;
+  let state: StateType;
+
+  // Set up generic axios response; to be used for all tests.
+  // We only need to include the ID, NAME, TITLE and VISIT_ID
+  // as those are the entity fields which the breadcrumb looks for
+  // when requesting information from the API.
+  (axios.get as jest.Mock).mockImplementation(() =>
+    Promise.resolve({
+      data: {
+        ID: 1,
+        NAME: 'INVESTIGATION 1',
+        TITLE: 'Test 1',
+        VISIT_ID: '1',
+      },
+    })
+  );
+
+  const createWrapper = (state: StateType): ReactWrapper => {
+    const mockStore = configureStore([thunk]);
+    return mount(
+      <Provider store={mockStore(state)}>
+        <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
+          <PageBreadcrumbs />
+        </MemoryRouter>
+      </Provider>
+    );
+  };
+
+  // Ensure that we can flush all promises before updating a wrapper.
+  const flushPromises = (): Promise<NodeJS.Immediate> =>
```

Comment:
This type looks overly complex - it can probably just be Promise<void>

Reply:
@louise-davies replaced with Promise<void> as you suggested in https://github.com/ral-facilities/datagateway/pull/70/commits/93f854af36b7327f17c63fc1602ea46bf0ccb66b.